common/BUILD

```diff
@@ -67,25 +87,29 @@ java_plugin(
     name = "dagger_compiler_plugin",
     generates_api = True,
     processor_class = "dagger.internal.codegen.ComponentProcessor",
+    tags = ["checkstyle_ignore"],
```

Comment:
What is the difference between `tags = ["checkstyle_ignore"],` and `allow_failure = 1,`? Why are some targets like this and others like that?

Reply:
@oferb I'll put a description of how it all works into `tools/checkstyle/README.md` in next commit

Reply:
@oferb fixed in https://github.com/google/startup-os/pull/486/commits/70e8e9de1d93cdafea77bd4488c6318c5c648a36